src/Illuminate/Auth/Access/Gate.php

```diff
@@ -739,7 +739,7 @@ protected function guessPolicyName($class)
     /**
      * Specify a callback to be used to guess policy names.
      *
-     * @param  callable  $callback
+     * @param  callable(class-string): string|string[]  $callback
```

Comment:
In assume, you need parentheses here, so that it's not ambiguous whether the return value is `string|string` or the param type is `callable|string[]`:
```suggestion
     * @param  (callable(class-string): string)|string[]  $callback
```